src/jit/lower.cpp

```diff
@@ -2875,8 +2875,9 @@ void Lowering::InsertPInvokeMethodProlog()
     firstBlockRange.InsertBefore(insertionPoint, LIR::SeqTree(comp, storeSP));
     DISPTREERANGE(firstBlockRange, storeSP);
 
-#endif // !_TARGET_X86_
+#endif // !defined(_TARGET_X86_) && !defined(_TARGET_ARM_)
 
+#if !defined(_TARGET_ARM_)
     // --------------------------------------------------------
```

Comment:
New output without that `#ifdef` adds a store not present in the original legacy output (which is [here](https://gist.githubusercontent.com/mskvortsov/bcdc1d731c8c7c72ead647267eb9802f/raw/1d0146a8f9fbe21d43d91d67fad73eff74318611/IL_STUB_PInvoke.legacy.txt)):
<pre>
G_M40565_IG02:
000014  AC02           add     r4, sp, 8
000016  F24D 0325      movw    r3, 0xd025
00001A  F2CB 53CA      movt    r3, 0xb5ca
00001E  4798           blx     r3               // CORINFO_HELP_INIT_PINVOKE_FRAME
<b>000020  4658           mov     r0, r11
000022  9007           str     r0, [sp+0x1c]</b>
000024  2000           movs    r0, 0
000026  2F00           cmp     r7, 0
000028  D002           beq     SHORT G_M40565_IG03
00002A  9709           str     r7, [sp+0x24]    // [V03 loc2]
00002C  F107 0008      add     r0, r7, 8

G_M40565_IG03:
000030  9B08           ldr     r3, [sp+0x20]    // [V04 tmp0]
000032  6A9B           ldr     r3, [r3+40]
000034  681B           ldr     r3, [r3]
000036  2400           movs    r4, 0
000038  2200           movs    r2, 0
00003A  9204           str     r2, [sp+0x10]    // [V07 PInvokeFrame+0x0c]
00003C  F20F 0209      adr     r2, G_M40565_IG05
000040  9206           str     r2, [sp+0x18]    // [V07 PInvokeFrame+0x14]
000042  2200           movs    r2, 0
000044  722A           strb    r2, [r5+8]

G_M40565_IG04:
000046  4798           blx     r3
</pre>
`genPInvokeMethodProlog` of `codegenlegacy.cpp` has the following fragment:
```c++
#if defined(_TARGET_X86_)
    if (threadTlsIndex == (DWORD)-1 || pInfo->osType != CORINFO_WINNT)
#else
    if (true)
#endif
    {
        ...
        // We are done, the rest of this function deals with the inlined case.
        return initRegs;
    }
```
therefore the code which sets `offsetOfCalleeSavedFP` near the end of the function doesn't get executed:
```c++
    /* set EBP value in frame */
    getEmitter()->emitIns_S_R(ins_Store(TYP_I_IMPL), EA_PTRSIZE, genFramePointerReg(),
                              compiler->lvaInlinedPInvokeFrameVar, pInfo->inlinedCallFrameInfo.offsetOfCalleeSavedFP);
```